public/src/components/channelManagement/epicTests/epicTestEditor.tsx

```diff
@@ -107,7 +107,10 @@ const EpicTestEditor: React.FC<EpicTestEditorProps> = ({
 
   const onVariantDelete = (deletedVariantName: string): void => {
     const updatedVariantList = test.variants.filter(variant => variant.name !== deletedVariantName);
-    onVariantsChange(updatedVariantList);
+    const controlProportionSettings =
+      updatedVariantList.length >= 2 ? test.controlProportionSettings : undefined;
```

Comment:
can we be a bit more explicit with something like this (pseudo typescript)
```suggestion
      updatedVariantList.exists(!_.equalsIgnoreCase("control")) ? test.controlProportionSettings : undefined;
```

Reply:
Yeahh this is a great point - actually we should probably check that there is also a variant left called `control`

Reply:
Perhaps (going all out)

```ts
  function hasControl(variants: EpicVariant[]): boolean {
    return variants.findIndex(v => v.name.toLowerCase() === 'control') > -1;
  }

  function hasAtLeastOneNonControlVariant(variants: EpicVariant[]): boolean {
    return variants.filter(v => v.name.toLowerCase() !== 'control').length >= 1;
  }

  const onVariantDelete = (deletedVariantName: string): void => {
    const updatedVariantList = test.variants.filter(variant => variant.name !== deletedVariantName);
    const controlProportionSettings =
      hasControl(updatedVariantList) && hasAtLeastOneNonControlVariant(updatedVariantList)
        ? test.controlProportionSettings
        : undefined;

    updateTest({ ...test, variants: updatedVariantList, controlProportionSettings });
  };
```

Or potentially that check could just be 

```ts
      hasControl(updatedVariantList) && updatedVariantList.length >= 2
```

which should also work

Reply:
sorry didn't spot this as it was "outdated" but yeah I duplicated your suggestion before I saw this :)  So go for it!

Reply:
I think the naming of the functions here is the really useful part, as it enables the intention/why of the code to be clear, rather than just the mechanics.